src/rhsm_debug/debug_commands.py

```diff
@@ -69,9 +69,20 @@ def _get_usage(self):
         return _("%%prog %s [OPTIONS] ") % self.name
 
     def _validate_options(self):
-        if self.options.destination and self.options.archive:
-            if not os.path.exists(self.options.destination):
+        self.assemble_path = self._get_assemble_dir()
+        if not os.path.isdir(self.assemble_path):
+            os.makedirs(self.assemble_path)
```

Comment:
I think _validate_options should not have any side effects (like create dirs or even instance variables if avoidable).

Since assemble_dir is just a hook for testing